osdu/client/aws.py

```diff
@@ -71,6 +72,16 @@ def get_tokens(self, password, secret_hash) -> None:
             AuthParameters=auth_params
         )
 
-
+        self._token_expiration = response['AuthenticationResult']['ExpiresIn'] + time()
         self._access_token = response['AuthenticationResult']['AccessToken']
         self._refresh_token = response['AuthenticationResult']['RefreshToken']
+    
+    # TODO: refresh can only be used if password is in environment variables. Is there another way to store the password securely?
+    def _update_token(self):
+        password = os.environ.get('OSDU_PASSWORD')
+        if(password):
+            self.get_tokens(password, self._secret_hash)
+            password = None
+            return self._access_token, self._token_expiration
+        return None # If we don't have a password, we can't refresh the token with the AWS client
```

Comment:
> # If we don't have a password, we can't refresh the token with the AWS client
Should this throw an Exception then?

Reply:
We should--that would bring it in line with what happens on the simple client if any of the env variables are missing.
Added.